java/src/jmri/jmrit/logixng/implementation/DefaultNamedTableManager.java

```diff
@@ -149,7 +160,12 @@ public AnonymousTable newAnonymousTable(int numRows, int numColumns)
     public NamedTable loadTableFromCSVData(
             @Nonnull String sys, @CheckForNull String user, @Nonnull String text)
             throws NamedBean.BadUserNameException, NamedBean.BadSystemNameException {
-        return AbstractNamedTable.loadTableFromCSV_Text(sys, user, text, true);
+        try {
+            return AbstractNamedTable.loadTableFromCSV_Text(sys, user, text, true, Table.CsvType.TABBED);
+        } catch (IOException e) {
+            log.error("Load from table failed with an IOException", e);
+            return null;
+        }
```

Comment:
This change must be reverted. This exception should not be handled here.

You need to:
1. Change the interface of the method to throw `IOException` instead of `BadUserNameException` and `BadSystemNameException`.
2. You need to change the method `NamedTableManager.loadTableFromCSVData()` in the same way.
3. Update Javadoc for `NamedTableManager.loadTableFromCSVData()` with:
    `* @throws java.io.IOException in case of an exception`
4. Update ReferenceUtilTest so that the methods `setupTables()` and `setUp()` throws IOException.

Reply:
I think I've done this the way you requested, but I'm not sure, please verify.